confit: replace with a re-export of beetbox/confuse

arcresu opened the pull request:
This re-exports all of the [confuse](https://github.com/sampsyo/confuse) module under `beets.util.confit` and patches in the minimum subset of commits to confit that have not yet been ported to confuse such that the test suite passes. A warning is issued when importing from the deprecated namespace.

Remaining work:
  - [x] Port the outstanding commits from confit to confuse (sampsyo/confuse#48, sampsyo/confuse#49, sampsyo/confuse#50).
  - [x] Make a new release of confuse.
  - [x] Update the minimum dependency of beets on confuse to that new release
  - [x] Remove the patches added here (including the change to flake8 config).
  - [x] Remove all uses of the deprecated namespace from beets itself (this should be left until later so as to avoid picking up many conflicts in this branch while the previous steps happen).
  - [x] Add changelog entry.
  - [x] Check if there are any irrelevant tests and documentation (including the wiki).
  - [ ] Notify downstream packagers about the new dependency (although I don't think that we need to block a beets release on this as discussed in https://github.com/beetbox/beets/pull/2061#issuecomment-480565229).

Supersedes and closes #2061. Part of #1966.

sampsyo commented:
Wow!! This is truly hard work, but very important despite being user-invisible. Thank you for taking this on! :horse:

It looks like we're pretty close to having the Confuse repository in sync. I think we should do this in the next release.

arcresu commented:
There are no currently-open PRs that would be affected by pulling out Confit (other than the older modularisation PR).

---
Here's the command to check in future (relies on having [PRs checked out locally](https://stackoverflow.com/a/43019825/759418)):
```
git for-each-ref --no-merged=upstream/master refs/remotes/upstream/pr/ --format='%(refname)' | while read r; do if ! git diff --quiet upstream/master...$r -- beets/util/confit.py; then echo ${r##*/}; fi; done | sort -n | sed -e's/^/- [ ] #/g'
```

sampsyo commented:
I'd love to address this next as part of the modularization plan! I've submitted Confuse v1.0.0 to PyPI (and tagged it on GitHub), so the next step seems to be to try depending that in our own setup.py.

arcresu commented:
Great! Thanks for taking care of yet another release :) I've updated the dependency on confuse and removed the backports. The tests pass locally for me and if it also looks good on the CI I'll make the additional changes needed to finish off this PR.

arcresu commented on beets/ui/commands.py:
```diff
@@ -39,7 +39,7 @@
 from beets import library
 from beets import config
 from beets import logging
-from beets.util.confit import _package_path
```
This has been importing the internal `_package_path` function. Now that confuse is a separate library this seems like the wrong thing to do. Either that functionality should be exposed publicly in confuse or we should find another way to do this.

sampsyo replied:
Indeed. The (sole) use of this could probably be easily replaced with a more standard `os.path.dirname(__file__)` sort of construction.

arcresu replied:
Great, then I'll sort this small change out separately and we can merge this now!

arcresu commented:
Ok, I think we're good to go! :tada:
